menus: make menus that reference unregistered commands more robust

vince-fugnitto opened the pull request:
<!--
Thank you for your Pull Request. Please provide a description and review
the requirements below.

Contributors guide: https://github.com/theia-ide/theia/blob/master/CONTRIBUTING.md
-->

<!--
Note: Security vulnerabilities should not be disclosed on GitHub, through a PR or any
other means. See SECURITY.md at the root of this repository, to learn how to report
vulnerabilities.
-->

#### What it does
<!-- Include relevant issues and describe how they are addressed. -->

Fixes: https://github.com/eclipse-theia/theia/issues/9870

The pull-request updates the registration of menus that reference invalid or missing commands to be more robust.
Rather than fail the layout, or the start of the application, the menus items will be omitted with an error log for extenders to handle.

#### How to test
<!-- Explain how a reviewer can reproduce a bug, test new functionality or verify performance improvements. -->

1. start the application, and confirm it starts correctly
2. the menus should be present - there should be error logs in the console regarding missing commands
3. repeat the steps for the browser or electron target respectfully.

https://user-images.githubusercontent.com/40359487/129045597-adcc19b1-699e-4e36-845e-6d2b525d50ca.mp4

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

msujew reviewed (approved):
I can confirm the issue is present on `master` and is addressed by this change nicely:

* On `master` unregistering those commands crashes the application on startup ❌ 
* With this change in place, the application starts up nicely ✔️ 
* The menu entries related to the commands are not visible ✔️
* The missing commands are logged as errors in the console (perhaps a warning would suffice, but I don't really care) ✔️

vince-fugnitto commented:
I updated the pull-request following @paul-marechal's offline comments about having a more permanent way to verify the feature in the framework (previously a dropped commit). I added a api-test, as well as a sample contribution in `api-samples` to both verify the behavior when a menu with an invalid command is registered.

paul-marechal reviewed (approved):
LGTM, the application is not blocked because of unknown commands in menus anymore.
